haystack_experimental/dataclasses/breakpoints.py

```diff
@@ -69,60 +69,26 @@ def __repr__(self):
 class AgentBreakpoint:
     """
     A dataclass to hold a breakpoint that can be used to debug an Agent.
-
-    It holds a set of breakpoints for the components in the Agent.
     """
 
-    generator_breakpoints: Set[Breakpoint]
-    tool_breakpoints: Set[ToolBreakpoint]
-
-    def __init__(self, breakpoints: Set[Union[Breakpoint, ToolBreakpoint]]):
-        if any(isinstance(bp, Breakpoint) and bp.component_name != "chat_generator" for bp in breakpoints):
-            raise ValueError("All Breakpoints must have component_name 'chat_generator'.")
-
-        if any(isinstance(bp, ToolBreakpoint) and bp.component_name != "tool_invoker" for bp in breakpoints):
-            raise ValueError("All ToolBreakpoints must have component_name 'tool_invoker'.")
-
-        if not breakpoints:
-            raise ValueError("Breakpoints must be provided.")
-
-        self.tool_breakpoints = set()
-        self.generator_breakpoints = set()
+    agent_name: str = ""
+    generator_breakpoint: Optional[Breakpoint] = None
+    tool_breakpoint: Optional[ToolBreakpoint] = None
 
-        for break_point in breakpoints:
-            if isinstance(break_point, Breakpoint):
-                self.generator_breakpoints.add(break_point)
-            elif isinstance(break_point, ToolBreakpoint):
-                self.tool_breakpoints.add(break_point)
-            else:
-                raise ValueError("Breakpoints must be either Breakpoint or ToolBreakpoint.")
+    def __init__(self, agent_name: str, break_point: Union[Breakpoint, ToolBreakpoint]):
+        if not isinstance(break_point, ToolBreakpoint) and break_point.component_name != "chat_generator":
+            raise ValueError("A Breakpoint must have component_name 'chat_generator'.")
 
-    def add_breakpoint(self, break_point: Union[Breakpoint, ToolBreakpoint]) -> None:
-        """
-        Adds a breakpoint to the set of breakpoints.
-        """
+        if not break_point:
+            raise ValueError("A Breakpoint must be provided.")
 
-        if isinstance(break_point, Breakpoint):
-            if break_point in self.generator_breakpoints:
-                raise ValueError(f"Breakpoint {break_point} already exists in generator breakpoints.")
-            self.generator_breakpoints.add(break_point)
+        self.agent_name = agent_name
+        self.tool_breakpoint = None
+        self.generator_breakpoint = None
```

Comment:
Just a general question. If we want to restrict users to only be able to pass one breakpoint to AgentBreakpoint, do you think it would make sense to update the dataclass to be something like
```python
class AgentBreakpoint:
    """
    A dataclass to hold a breakpoint that can be used to debug an Agent.
    """
    breakpoint: Union[Breakpoint, ToolBreakPoint]
    agent_name: str = ""
```
I think this would help simplify the `__init__` logic.

Reply:
agree 👍🏽 - I will update the code accordingly